### ISSUE TITLE
Batch starts twice

### ISSUE THREAD

**Issue by amine-labassi** [BatchLabs/ios-sdk#22]
Batch starts twice : 
[Batch] Batch is already started, restarting again.

env : 
*   dependencies
  pod 'GoogleTagManager', '~> 6.0'
  pod 'Batch', '~> 1.14'
*  iphone 8 iOS 13;3

**Comment by abarisain** [comment]
Hello!

Do you call startWithAPIKey: only once in application didFinishLaunchingWithOptions?

If so, do you use UIScene in your app?

**Comment by amine-labassi** [comment]
Hello,

Yes i do call startWithAPIKey only once :
```swift
@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

    var notificationDelegate = NotificationDelegate()
    var center = UNUserNotificationCenter.current()
    var window: UIWindow?

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        
        // BatchPush.disableAutomaticIntegration()
        Batch.start(withAPIKey: "*************") // dev
        // Batch.start(withAPIKey: "*************") // live
        BatchPush.requestNotificationAuthorization()
        BatchPush.refreshToken()
        
        self.center.delegate = self.notificationDelegate
        
        // FirebaseApp.configure()
        return true
    }
```

Concerning UIScene, i'am not using it explicitly but i have a SceneDelegate class so maybe it's used behind the scene :
```swift 
@available(iOS 13.0, *)
class SceneDelegate: UIResponder, UIWindowSceneDelegate {

    var window: UIWindow?
}
```
Thank you for your help,

Regards,

**Comment by abarisain** [comment]
Thanks.

Knowing if your app uses multiple scenes can be done by checking "supports multiple windows"
![image](https://user-images.githubusercontent.com/312529/74102146-24c3f180-4b41-11ea-9353-a74c74c23895.png)


If so, this is a known bug.

**Comment by amine-labassi** [comment]
thank you for the quick reply.
The "Supports multiple windows" is not checked in my target config.

![image](https://user-images.githubusercontent.com/16447595/74102345-0e1e9a00-4b43-11ea-9a0e-44caad8f926f.png)

**Comment by abarisain** [comment]
Thank you. I still think it's related to your scenedelegate, but that gives us more context!

Anyway this log is harmless, we plan to fix this cosmetic issue in 1.16

**Comment by amine-labassi** [comment]
Unfortunately, it's not only a cosmetic issue as it provokes landing pages to popin twice.

**Comment by amine-labassi** [comment]
by the way, is it also normal that no handler is fired when receiving a and notif event, but the landing popin in showing ?

```swift 
@objc
class NotificationDelegate: NSObject, UNUserNotificationCenterDelegate {
    
    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
        BatchPush.handle(userNotificationCenter: center, didReceive: response)
    }
    
    func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
        BatchPush.handle(userNotificationCenter: center, willPresent: notification, willShowSystemForegroundAlert: true)
        completionHandler([.alert, .sound, .badge])
    }
}
```

**Comment by abarisain** [comment]
>Unfortunately, it's not only a cosmetic issue as it provokes landing pages to popin twice.

Landing or In-App messages? I believe I can produce a workaround for the first case.

If your app doesn't support multi window, the quickest fix is to remove all traces of UIScene from your app: delete the delegate, and remove the scene entries in Info.plist.

>by the way, is it also normal that no handler is fired when receiving a notif event, but the landing popin in showing ?

That looks like another issue. Can you put a breakpoint and see if it hits? Note that if it is another issue, please open a ticket on our support. Please tell the support team I sent you from the issue.

**Comment by amine-labassi** [comment]
I don't know exaclty what the account manager did configure as compaign.
But it's maybe InApp Messages as i see in logs "BAMSGInterstitialViewController".
BAM stands for **B**atch in**A**pp **M**essage ?

I will remove the UIScene totally from my app and let you know if once the issue is resolved.

**Comment by amine-labassi** [comment]
Perfect ! i removed UIScene and now Batch is starting once.
Otherwise, i put a breakpoint on NotificationDelegate sub-class but the debugger doesn't hits.

**Comment by abarisain** [comment]
Glad it fixed it!

Please note that these breakpoints will only be hit when:
- opening a push notification
- receiving a push notification while your app is in the foreground 


I'll close this issue as we're tracking UIScene bugs internally.
Thanks,
